src/prototype.rs

```diff
@@ -318,6 +429,26 @@ impl<'a> FirstPass<'a> {
         }
     }
 
+
+    /// Appends a line of HTML to the tree.
+    fn append_html_line(&mut self, start: usize, end: usize)
+    {
+
+        if end > start {
+            self.tree.append(Item {
+                start: start,
+                // Normalize CRLF to LF
+                end: if self.text.as_bytes()[end - 2] == b'\r' { end - 2 } else { end - 1},
+                body: ItemBody::Html,
+            });
+        }
+        self.tree.append(Item {
```

Comment:
If there's a newline in the source, I prefer to output it as text rather than synthesizing a new line, for sourcemap purposes. I suspect we should probably synthesize a newline in the case where the input ends without a newline.

Possibly some of this logic can be combined with the code case, but I'm not too worried, a little duplication is fine.

Reply:
I was just trying to mimic the existing `fn append_html_line()` (impl on `Tree`). I assumed it was done for a reason over there.  Looks like I should have been mimicking `append_code_text` instead!

Reply:
Yeah, `append_code_text` was done after the infrastructure was put in place for the fun line start handling with tabs.